In [1]:
import numpy as np # linear algebra
from scipy import stats
from scipy.stats import norm

# **Problem Statement 1**:

The marks awarded for an assignment set for a Year 8 class of 20 students were as
follows:
6 7 5 7 7 8 7 6 9 7 4 10 6 8 8 9 5 6 4 8

In [2]:
marks = [6,7,5,7,7,8,7,6,9,7,4,10,6,8,8,9,5,6,4,8]
print("Mean : ",np.mean(marks))
print("Median : ",np.median(marks))
print("Mode : ",stats.mode(marks))
print("Standard Deviation : ",np.std(marks))

Mean :  6.85
Median :  7.0
Mode :  ModeResult(mode=array([7]), count=array([5]))
Standard Deviation :  1.5898113095584647


# **Problem Statement 2:**

The number of calls from motorists per day for roadside service was recorded for a
particular month:
28, 122, 217, 130, 120, 86, 80, 90, 140, 120, 70, 40, 145, 113, 90, 68, 174, 194, 170,
100, 75, 104, 97, 75, 123, 100, 75, 104, 97, 75, 123, 100, 89, 120, 109


In [3]:
calls=[28, 122, 217, 130, 120, 86, 80, 90, 140, 120, 70, 40, 145, 113, 90, 68, 174, 194, 170, 100, 75, 104, 97, 75, 123, 100, 75, 104, 97, 75, 123, 100, 89, 120, 109]
print("Mean : ",np.mean(calls))
print("Median : ",np.median(calls))
print("Mode : ",stats.mode(calls))
print("Standard Deviation : ",np.std(calls))

Mean :  107.51428571428572
Median :  100.0
Mode :  ModeResult(mode=array([75]), count=array([4]))
Standard Deviation :  38.77287080168403


# **Problem Statement 3:**
The number of times I go to the gym in weekdays, are given below along with its
associated probability:
x = 0, 1, 2, 3, 4, 5
f(x) = 0.09, 0.15, 0.40, 0.25, 0.10, 0.01
Calculate the mean no. of workouts in a week. Also evaluate the variance involved in
it

In [4]:
x=[0, 1, 2, 3, 4, 5]
f_x=[0.09, 0.15, 0.40, 0.25, 0.10, 0.01]
workouts=np.array(x)*f_x
print("Mean no. of workouts: ",norm.fit(workouts)[0])
print("Variance: ",np.var(workouts))

Mean no. of workouts:  0.35833333333333334
Variance:  0.10284722222222224


# **Problem Statement 5:**

A company manufactures LED bulbs with a faulty rate of 30%. If I randomly select 6
chosen LEDs, what is the probability of having 2 faulty LEDs in my sample?
Calculate the average value of this process. Also evaluate the standard deviation
associated with it.

In [5]:
# Probability of having 2 faulty LEDs

from scipy.stats import binom
import matplotlib.pyplot as plt
import seaborn as sns
probab=binom.pmf(k=2,n=6,p=0.3)
print("Probability will be :",probab)

# Mean and standard deviation

import math
mean,var=binom.stats(n=6,p=0.3)
print("mean := ",mean)
print("standard deviation :=",math.sqrt(var))

Probability will be : 0.32413499999999995
mean :=  1.7999999999999998
standard deviation := 1.1224972160321822


# **Problem Statement 6:**

Gaurav and Barakha are both preparing for entrance exams. Gaurav attempts to
solve 8 questions per day with a correction rate of 75%, while Barakha averages
around 12 questions per day with a correction rate of 45%. What is the probability
that each of them will solve 5 questions correctly? What happens in cases of 4 and 6
correct solutions? What do you infer from it? What are the two main governing
factors affecting their ability to solve questions correctly? Give a pictorial
representation of the same to validate your answer.

In [6]:
# Probability of Gaurav solving 5 questions correctly
questions =[5,4,6]
for q in questions:
    print("Probability of Gaurav solving " + str(q) + " questions correctly is :",binom.pmf(k=q,n=8,p=0.75))
    print("Probability of Barkha solving " + str(q) + " questions correctly is :",binom.pmf(k=q,n=12,p=0.45))
    print("----------------------------------------------------------------------------")


Probability of Gaurav solving 5 questions correctly is : 0.20764160156250022
Probability of Barkha solving 5 questions correctly is : 0.22249823843265812
----------------------------------------------------------------------------
Probability of Gaurav solving 4 questions correctly is : 0.08651733398437506
Probability of Barkha solving 4 questions correctly is : 0.1699639321360581
----------------------------------------------------------------------------
Probability of Gaurav solving 6 questions correctly is : 0.31146240234375017
Probability of Barkha solving 6 questions correctly is : 0.21238468214026432
----------------------------------------------------------------------------


The inference that can be drawn from this is that while Gaurav has a high correction rate, it is based on only 8 questions per day. Therefore the true picture can only be seen when the number of questions solved correctly is higher (5 or 6)

# **Problem Statement 7:**

Customers arrive at a rate of 72 per hour to my shop. What is the probability of 𝑘
customers arriving in 4 minutes? a) 5 customers, b) not more than 3 customers, c)
more than 3 customers. Give a pictorial representation of the same to validate your
answer.

In [7]:
# Calculating Mu with the time interval of 4 minutes

mu=72*4/60

# Calculating probabilities

from scipy.stats import poisson
print("The probability of 5 customers arriving in 4 minutes: ", poisson.pmf(k=5,mu=mu))
print("The probability of not more than 3 customers arriving in 4 minutes: ", sum(poisson.pmf(k=[0,1,2],mu=mu)))
print("The probability of more than 3 customers arriving in 4 minutes: ", 1-sum(poisson.pmf(k=[0,1,2],mu=mu)))


The probability of 5 customers arriving in 4 minutes:  0.17474768364388296
The probability of not more than 3 customers arriving in 4 minutes:  0.14253921888902693
The probability of more than 3 customers arriving in 4 minutes:  0.8574607811109731


# **Problem Statement 8:**

I work as a data analyst in Aeon Learning Pvt. Ltd. After analyzing data, I make
reports, where I have the efficiency of entering 77 words per minute with 6 errors per
hour. What is the probability that I will commit 2 errors in a 455-word financial report?
What happens when the no. of words increases/decreases (in case of 1000 words,
255 words)?
How is the 𝜆 affected?
How does it influence the PMF?
Give a pictorial representation of the same to validate your answer.

In [8]:
# Calculating probabilities
mu1=6*455/(77*60)
mu2=6*1000/(77*60)
mu3=6*255/(77*60)
print("The probability of of commiting 2 errors in a 455-word report: ", poisson.pmf(k=2,mu=mu1))
print("The probability of of commiting 2 errors in a 1000-word report:  ", poisson.pmf(k=2,mu=mu2))
print("The probability of of commiting 2 errors in a 255-word report:  ", poisson.pmf(k=2,mu=mu3))


The probability of of commiting 2 errors in a 455-word report:  0.09669027375144444
The probability of of commiting 2 errors in a 1000-word report:   0.23012815007300153
The probability of of commiting 2 errors in a 255-word report:   0.039377135392854104


# **Problem Statement 10:**

Please compute the following:
a) P(Z > 1.26), . P(Z < −0.86), P(Z > −1.37), P(−1.25 < Z < 0.37), . P(Z ≤ −4.6)
b) Find the value 𝑧 such that 𝑃(𝑍 > 𝑧) = 0.05
c) Find the value of 𝑧 such that 𝑃(−𝑧 < 𝑍 < 𝑧) = 0.99

In [9]:
# a) P(Z > 1.26), . P(Z < −0.86), P(Z > −1.37), P(−1.25 < Z < 0.37), . P(Z ≤ −4.6)

print("P(Z>1.26): ",1-stats.norm.cdf(1.26))
print("P(Z<-0.86): ",stats.norm.cdf(-0.86))
print("P(Z>-1.37): ",1-stats.norm.cdf(-1.37))
print("P(-1.25<Z<0.37): ",stats.norm.cdf(0.37)-stats.norm.cdf(-1.25))
print("P(Z≤ −4.6): ",stats.norm.cdf(-4.6))

# b) Find the value 𝑧 such that 𝑃(𝑍 > 𝑧) = 0.05

print("The value of z such that P(Z>z)=0.05: ", stats.norm.ppf(0.95))

P(Z>1.26):  0.10383468112130045
P(Z<-0.86):  0.1948945212518084
P(Z>-1.37):  0.914656549178033
P(-1.25<Z<0.37):  0.5386589811336914
P(Z≤ −4.6):  2.1124547025028537e-06
The value of z such that P(Z>z)=0.05:  1.6448536269514722


# **Problem Statement 11:**

The current flow in a copper wire follow a normal distribution with a mean of 10 𝑚A
and a variance of 4 (𝑚𝐴)2.
What is the probability that a current measurement will exceed 13 𝑚𝐴? What is the
probability that a current measurement is between 9 and 11mA? Determine the
current measurement which has a probability of 0.98.

In [10]:
x = 13
mean_current = 10
variance_current = 4
sd_current=np.sqrt(variance_current)
prob=1-stats.norm.cdf((x-mean_current)/sd_current)
print(' Probability that a current measurement will exceed 13mA is {:.2f}%'.format(prob*100))

x1 = 9
x2 = 11
prob1=stats.norm.cdf((x2-mean_current)/sd_current)- stats.norm.cdf((x1-mean_current)/sd_current)
print(' Probability that a current measurement is between 9 and 11 mA is {:.2f}%'.format(prob1*100))
print(' The current measurement which has a probability of 0.98 is {:.2f} mA'.format(stats.norm.ppf(0.98)*sd_current+mean_current))


 Probability that a current measurement will exceed 13mA is 6.68%
 Probability that a current measurement is between 9 and 11 mA is 38.29%
 The current measurement which has a probability of 0.98 is 14.11 mA


# **Problem Statement 12:**

The shaft in a piston has its diameter normally distributed with a mean of 0.2508 inch
and a standard deviation of 0.0005 inch. The specifications of the shaft are 0.2500 ∓
0.0015 inch. What proportion of shafts are in sync with the specifications? If the
process is centered so that the mean is equal to the target value of 0.2500, what
proportion of shafts conform to the new specifications? What is your conclusion from
this experiment?

In [11]:
mean_diameter = 0.2508
std_diameter= 0.0005
prob=stats.norm.cdf((0.2515-mean_diameter)/std_diameter)- stats.norm.cdf((0.2485-mean_diameter)/std_diameter)
print('Proportion of shafts in sync: {:.2f}%'.format(prob*100))
prob1=stats.norm.cdf((0.2515-0.25)/std_diameter)- stats.norm.cdf((0.2485-0.25)/std_diameter)
print('Proportion of shafts in sync if mean diameter is 0.25: {:.2f}%'.format(prob1*100))

Proportion of shafts in sync: 91.92%
Proportion of shafts in sync if mean diameter is 0.25: 99.73%


Most of the nonconforming shafts are too large, because the process mean is located very near to the upper specification limit. If the process is centered so that the process mean is equal to the target value of 0.2500,, the yield is increased to approximately 99.73%


In [12]:
stats.norm.cdf(1.98)-stats.norm.cdf(-1.98)

0.952296471316983